spacy/pipeline/lemmatizer.py

```diff
@@ -0,0 +1,277 @@
+        "model": None,
+        "mode": "lookup",
+        "lookups": None,
+        "overwrite": False,
+    },
+    scores=["lemma_acc"],
+    default_score_weights={"lemma_acc": 1.0},
+)
+def make_lemmatizer(
+    nlp: Language,
+    model: Optional[Model],
+    name: str,
+    mode: str,
+    lookups: Optional[Lookups],
+    overwrite: bool = False,
+):
+    lookups = Lemmatizer.load_lookups(nlp.lang, mode, lookups)
+    return Lemmatizer(
+        nlp.vocab, model, name, mode=mode, lookups=lookups, overwrite=overwrite
+    )
+
+
+class Lemmatizer(Pipe):
+    """
+    The Lemmatizer supports simple part-of-speech-sensitive suffix rules and
+    lookup tables.
+
+    DOCS: https://spacy.io/api/lemmatizer
+    """
+
+    @classmethod
+    def get_lookups_config(cls, mode: str) -> Dict:
+        if mode == "lookup":
+            return {
+                "required_tables": ["lemma_lookup"],
+            }
+        elif mode == "rule":
+            return {
+                "required_tables": ["lemma_rules"],
+                "optional_tables": ["lemma_exc", "lemma_index"],
+            }
+        return {}
+
+    @classmethod
+    def load_lookups(
+        cls,
+        lang: str,
+        mode: str,
+        lookups: Optional[Lookups],
+    ) -> Lookups:
+        config = cls.get_lookups_config(mode)
+        required_tables = config.get("required_tables", [])
+        optional_tables = config.get("optional_tables", [])
+        if lookups is None:
+            lookups = load_lookups(lang=lang, tables=required_tables)
+            optional_lookups = load_lookups(
+                lang=lang, tables=optional_tables, strict=False
+            )
+            for table in optional_lookups.tables:
+                lookups.set_table(table, optional_lookups.get_table(table))
+        for table in required_tables:
+            if table not in lookups:
+                raise ValueError(
+                    Errors.E1004.format(
+                        mode=mode, tables=required_tables, found=lookups.tables
+                    )
+                )
+        return lookups
+
+    def __init__(
+        self,
+        vocab: Vocab,
+        model: Optional[Model],
+        name: str = "lemmatizer",
+        *,
+        mode: str = "lookup",
+        lookups: Optional[Lookups] = None,
+        overwrite: bool = False,
+    ) -> None:
+        """Initialize a Lemmatizer.
+
+        vocab (Vocab): The vocab.
+        model (Model): A model (not yet implemented).
+        name (str): The component name. Defaults to "lemmatizer".
+        mode (str): The lemmatizer mode: "lookup", "rule". Defaults to "lookup".
+        lookups (Lookups): The lookups object containing the (optional) tables
+            such as "lemma_rules", "lemma_index", "lemma_exc" and
+            "lemma_lookup".
+        """
+        self.vocab = vocab
+        self.model = model
+        self._mode = mode
+        self.lookups = lookups if lookups is not None else Lookups()
```

Comment:
Just to double-check, will we be okay to modify a `Lookups` object that's passed in here? For instance, do we ever have one in a global scope, e.g. in tests?

Reply:
I think so, but I don't really know for sure. I don't really understand the scope of registry stuff, but I moved all the lookups assets in the tests within the methods.

Reply:
I did run into problems with global assets in the tests because I had two things with the same name that resolved to one of them and broke the other test.